### ISSUE TITLE
Anything I select in the GUI just reloads the UI, does nothing else.

### ISSUE THREAD

**Issue by ktzee** [murkl/d2launcher#24]
I used to be able to use d2launcher fine and play MXL. I just reinstalled it on a fresh Arch but it doesn't work anymore.
The GUI opens fine, but whatever I choose it just reloads the UI and shows me the main menu again, it doesn't actually do anything. I tried running from terminal to see if it would output any error, but it doesn't output anything.
I tried downgrading zenity to a version of another user who has it working in Discord, but it didn't help.

Any idea of what it could be?

**Comment by murkl** [comment]
Thanks for reporting this. I can reproduce the same behavior under Arch. Strange, I'll investigate the problem...

**Comment by julchiar** [comment]
Hello, I'm on Arch with KDE and just installed d2launcher from the AUR. The UI doesn't seem to load at all, not sure what I'm doing. Launching d2launcher from console prints "no launcher update available", followed by a system notification of the new Median XL 2.10 update being available (can't interact with it). I don't see any UI or tray icon or anything of the sorts anywhere. The directory .d2launcher and config file were created in my home folder. 

I then tried running d2launcher with sudo and got the following:

> Added d2_dir="/root/.d2launcher/bin/diablo2" to /root/.d2launcher/d2launcher.conf
> Added d2_exe="Diablo II.exe" to /root/.d2launcher/d2launcher.conf
> Added d2_args="-3dfx" to /root/.d2launcher/d2launcher.conf
> Added gui_width="450" to /root/.d2launcher/d2launcher.conf
> Added gui_height="380" to /root/.d2launcher/d2launcher.conf
> Added update_check="true" to /root/.d2launcher/d2launcher.conf
> Added mxl_update_check="true" to /root/.d2launcher/d2launcher.conf
> Added mxl_update_channel="public" to /root/.d2launcher/d2launcher.conf
> Added d2_stats_tray="true" to /root/.d2launcher/d2launcher.conf
> No D2Launcher update available
> Error spawning command line “dbus-launch --autolaunch=7b75f0ccb0234cb9b5edceef824f9fd6 --binary-syntax --close-stderr”: Child process exited with code 1

On repeated sudo launches only the bottom two lines are displayed. Would love any assistance.

Also line 446 says "maninfest.json", might be a typo.

**Comment by murkl** [comment]
Sorry guys, I made a mistake. I just had the error during development and got it mixed up. With the latest commit on main branch I have improved the error handling a little bit and output error messages. Maybe that helps?
Does the error also occur if ~/.d2launcher dir has been deleted or moved?

**Comment by murkl** [comment]
> Hello, I'm on Arch with KDE and just installed d2launcher from the AUR. The UI doesn't seem to load at all, not sure what I'm doing. Launching d2launcher from console prints "no launcher update available", followed by a system notification of the new Median XL 2.10 update being available (can't interact with it). I don't see any UI or tray icon or anything of the sorts anywhere. The directory .d2launcher and config file were created in my home folder.
> 
> I then tried running d2launcher with sudo and got the following:
> 
> > Added d2_dir="/root/.d2launcher/bin/diablo2" to /root/.d2launcher/d2launcher.conf
> > Added d2_exe="Diablo II.exe" to /root/.d2launcher/d2launcher.conf
> > Added d2_args="-3dfx" to /root/.d2launcher/d2launcher.conf
> > Added gui_width="450" to /root/.d2launcher/d2launcher.conf
> > Added gui_height="380" to /root/.d2launcher/d2launcher.conf
> > Added update_check="true" to /root/.d2launcher/d2launcher.conf
> > Added mxl_update_check="true" to /root/.d2launcher/d2launcher.conf
> > Added mxl_update_channel="public" to /root/.d2launcher/d2launcher.conf
> > Added d2_stats_tray="true" to /root/.d2launcher/d2launcher.conf
> > No D2Launcher update available
> > Error spawning command line “dbus-launch --autolaunch=7b75f0ccb0234cb9b5edceef824f9fd6 --binary-syntax --close-stderr”: Child process exited with code 1
> 
> On repeated sudo launches only the bottom two lines are displayed. Would love any assistance.
> 
> Also line 446 says "maninfest.json", might be a typo.

Do not call with root, just “d2launcher” as user

**Comment by julchiar** [comment]
I did a clean install with the latest commits. It generated the .d2launcher folder as it should and then printed 

> Startup finished
> New D2Launcher 4.0.5 update available
> New Median XL 2.10.0 update available

After placing the d2 game folder into bin, a UI popped up, asking to create a backup. This completed successfully. Nothing else seems to happen afterwards, the main UI is still not showing up. The log file was also generated/saved this time but it just states the lines above.

**Comment by murkl** [comment]
Pretty strange. Can you please try again with the new commit? And does the following command work for you: `/usr/bin/zenity --info --text=“Version: $(/usr/bin/zenity --version)”`

**Comment by julchiar** [comment]
> Pretty strange. Can you please try again with the new commit? And does the following command work for you: `/usr/bin/zenity --info --text=“Version: $(/usr/bin/zenity --version)”`

> This option is not available. Please see --help for all possible usages.


The "--text" option doesn't work. Just running zenity --info creates a popup stating "All updates completed" and zenity --version says 4.0.3.

New d2launcher commit spams console output the error above but doesn't seem to do anything else.

**Comment by murkl** [comment]
But you started d2launcher without root, didn't you?
Is your dbus service running: `systemctlstatus dbus.service`?

**Comment by julchiar** [comment]
yes, without root.

```
systemctl status dbus.service
● dbus-broker.service - D-Bus System Message Bus
     Loaded: loaded (/usr/lib/systemd/system/dbus-broker.service; disabled; preset: disabled)
     Active: active (running) since Sat 2024-11-02 19:45:37 CET; 3h 21min ago
```

there's a few more lines that I omitted below

**Comment by murkl** [comment]
I suspect that something in your zenity_menu() function is not working properly. If you are confident, you can use echo to see how far it works for you.

**Comment by julchiar** [comment]
•

This one right there.

Lines 1125 and 1126 have this little dot (•) thing  and zenity does not seem to like it. Deleting those two makes the UI show up as intended.

**Comment by murkl** [comment]
Thanks for the test! I have replaced the special character to make it more compatible. Will be released in 4.1.0 soon.

**Comment by murkl** [comment]
Is released
